### PR TITLE
 Add docs for cancel API request 

### PR DESCRIPTION
**Pull request by Amplifiyer**
*Issue #, if available:* aws-amplify/amplify-js#5753

*Description of changes:*
Add docs for cancel API request . See aws-amplify/amplify-js#5753 for more details

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/lib/graphqlapi/fragments/js/cancel-request.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+API.cancel(promise, "my message for cancellation");
+```
+
+You need to ensure that the promise returned from `API.graphql()` has not been modified. Typically async functions wrap the promise being returned into another promise. For example, the following will not work
+
+```javascript
+async function makeAPICall() {
+    return API.graphql(graphqlOperation(...));
+}
+const promise = makeAPICall()
```
Comment (sammartinez) [inline docs/lib/graphqlapi/fragments/js/cancel-request.md:31]:
```suggestion
const promise = makeAPICall();

```

File: docs/lib/restapi/fragments/js/cancel.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+API.cancel(promise, "my message for cancellation");
+```
+
+You need to ensure that the promise returned from `API.get()` or any other API calls has not been modified. Typically async functions wrap the promise being returned into another promise. For example, the following will not work
+
+```javascript
+async function makeAPICall() {
+    return API.get(myApiName, myPath, myInit);
+}
+const promise = makeAPICall()
```
Comment (sammartinez) [inline docs/lib/restapi/fragments/js/cancel.md:31]:
```suggestion
const promise = makeAPICall();

```

File: docs/lib/restapi/fragments/js/cancel.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+## CANCEL API requests
```
Comment (sammartinez) [inline docs/lib/restapi/fragments/js/cancel.md:1]: Do we need two of the same files to be created? We can reuse the `cancel-request` fragment within this similar documentation. Let me know what you think

Comment (Amplifiyer) [reply docs/lib/restapi/fragments/js/cancel.md:1]: I was thinking about it, right now I have two different examples `API.get` and `API.graphql` in the two files. Also was wondering if cross referencing fragments was a good idea. If you feel it's better to reuse, I can make the change.

**Pull request merged**